src/include/hardware.h

```diff
@@ -73,6 +73,23 @@
 #define WT_WRITE_ONCE(v, val) WT_RELEASE_WRITE_WITH_BARRIER(v, val)
 #endif
 
+#ifdef HAVE_NEOVERSE
```

Comment:
I think it would be better to test the feature flag: `__ARM_FEATURE_RCPC` which should be 1.

Reply:
FWIW gcc 11.3.0 does support `__ARM_FEATURE_RCPC` as far as I can tell.

Reply:
The gcc 7 release https://gcc.gnu.org/gcc-7/changes.html added support for Armv8.3-a and apparently the assembler was updated, but not the compiler arch def -- which is unfortunate.

Reply:
Further investigation reveals that 11.3 doesn't support `__ARM_FEATURE_RCPC`. The two possibilities are:
1: Output ldapr by default for arm which is fine assuming we require everyone to be on armv8.3-a or
2: Move it behind HAVE_NEOVERSE.